lib/imgix/path.rb

```diff
@@ -131,7 +131,10 @@ def has_query?
     def build_srcset_pairs(params)
       srcset = ''
 
-      for width in @target_widths do
+      width_tolerance = params.delete('width_tolerance'.to_sym)
+      widths = width_tolerance == DEFAULT_WIDTH_TOLERANCE ? @target_widths : TARGET_WIDTHS.call(width_tolerance)
```

Comment:
@jayeb The sort of gotcha to all this is that we already calculate the default `TARGET_WIDTHS` when we instantiate a new `Path` object. But if the user opts for a different width tolerance, we end up recalculating a new set of widths on the fly. This is obviously going to cause us to perform repeated work whenever a user wants to specify their own tolerance, but in the grand scheme of things I still don't want to be calculating the widths **every** time we call `to_srcset`
I'm toying around with the idea of storing the custom widths locally to avoid recalculating on subsequent calls to `to_srcset(width_tolerance:x)`, but might just hold off until all other srcset features are built out.